content/building-blocks/automate-and-execute-your-work/reproducible-work/google_cloud_docker.md

```diff
@@ -0,0 +1,155 @@
+
+### Step 1: Install and Set up docker
+
+For this first step you can check our building block on [how to set up Docker.](https://tilburgsciencehub.com/building-blocks/configure-your-computer/automation-and-workflows/docker/)
+
+### Step 2: Build the environment's image from a Dockerfile
+
+Once you have Docker installed and working, the first thing you'll need to import your environment is a [Dockerfile](https://docs.docker.com/engine/reference/builder/). A Dockerfile provides Docker with the necessary indications to produce a Docker image, each of the instances of an image is known as a container. 
+
+{{% tip %}}
+
+In this case you can think of the Docker image as a model of our environment, and the Dockerfile as the manual on how to build this model. Each time you tell Docker to generate a new container (i.e. instance) of the image, this action would be equivalent to generate a new replica of the enviroment for us to use from the model you previously provided.
+
+{{% /tip %}}
+
+In order to tell Docker to build an image from the provided Dockerfile, open your command terminal in the directory were you have stored your dockerfile and run the following code:
+
+{{% codeblock %}}
+```bash
+# Build a Docker image 
+$ docker build .
+```
+{{% /codeblock %}}
+
+{{% warning %}}
+ 
+Do not forget to include the dot at the end of the code block above as it indicates Docker that it has look for your dockerfile in current directory.
+
+{{% /warning %}}
+
+{{% tip %}}
+ 
+You can  assign a name to your new Docker image by addign to the code above the flag "-t" followed by the name you want to assign in between of the build command and the dot at the end as shown in the code block below. 
+
+Don't worry if if you don't include this information as in that case docker will automatically assign a name to the image. However specifying a name is advisable as it will allow you to identify your images more easily. If you want to learn more about other options of the "build" command you can check [this link.](https://docs.docker.com/engine/reference/commandline/build/)
+
+{{% /tip %}}
+
+{{% codeblock %}}
+```bash
+# Build a Docker image and assign a name to it
+$ docker build -t your_image_name .
+```
+{{% /codeblock %}}
+
+### Step 3: Docker compose
+
+Docker compose is a tool within Docker whose main functionality is to coordinate your containers and provide them with the services required to run coherently. Although this can sound a bit confusing if you are new to Docker (You can learn more about Docker compose [here](https://docs.docker.com/compose/)), the key takeaway is that it will assist and simplify the task of replicating your environment by providing some extra information to Docker about how to do so.
+
+In the same way that you needed a dockerfile to provide Docker with the necessary instructions to build an image, a docker-compose file is required to tell docker how to provide these services to a (series of) container(s).

```

Comment:
`In the same way that you needed a dockerfile to provide Docker with the necessary instructions to build an image, a docker-compose file tells the docker engine how to build container(s) from the image.`

Also, maybe we can add a note to say that `docker run` command can theoretically be used to create and run containers and do what docker-compose does. But the advantage offered by docker-compose is that it can very smoothly create multiple containers, allow them to communicate with each other or start up containers with complicated settings. Also having all those instructions in a file (instead of typing it on the terminal) makes it easier to reproduce.